### PR TITLE
Implement degree planner UX

### LINKED ISSUE #65: [Feature Request] Implement Degree Planner Functionality
# Overview

The Degree Planner tool inside Nebula Planner is what students will use to plan out and customize the college experience they want to have. Currently, the Degree Planner has been partially implemented, but some things still need to be done before it is ready to be integrated into the rest of the project.

# Proposal

1. Create the ability to add courses 
2. Develop a method for users to search for UTD courses not part of their degree plan
3. Allow users to add/remove semesters 
4. Build a new plan creation flow

# Supplementary Information

The design of the Degree Planner will aim to follow mockups from the "Nebula Mockups" Figma file.

### PR DESCRIPTION
**Pull request by ZzRanger**
## Overview

Resolves #65 

The changes in this pull request flushed out basic functionality for the Degree Planner component. This component is now partially integrated with the rest of the application.

## What Changed

The following tasks were completed:

1. Create the ability to add courses 
2. Develop a method for users to search for UTD courses not part of their degree plan
3. Allow users to add/remove semesters
4. Build a new plan creation flow

## Future Steps

Notable features that need to be implemented are listed below.
1. Implementing the backend for the new plan creation flow
2. Get courses from Nebula API
3. Course generation after new plan creation flow (we need to integrate #55 here)
4. Store user plan data in Redux and external database

## Other Notes

This PR contains commits from branches 'feature/course-audit' and 'feature/new-onboarding' that have not been merged into the develop branch. Those branches and their respective PR's need to be addressed prior to merging this PR.

### CODE REVIEW

File: modules/common/store.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { useMemo } from 'react';
 import { createStore, applyMiddleware } from 'redux';
 import { composeWithDevTools } from 'redux-devtools-extension';
 import { persistReducer, persistStore } from 'redux-persist';
-import storage from 'redux-persist/lib/storage';
+//import storage from 'redux-persist/lib/storage';
```
Comment (CharlieMahana) [inline modules/common/store.ts:5]: Either remove this line of code or uncomment it

**Comment by CharlieMahana** [comment]
I noticed that in certain areas of the code, you've changed a horizontal JSON object expansion into a vertical JSON object expansion and vice versa. Make sure that you're linting and formatting your code to avoid problematic code and conform to a repo-wide style.

**Comment by WillieCubed** [comment]
> I noticed that in certain areas of the code, you've changed a horizontal JSON object expansion into a vertical JSON object expansion and vice versa. Make sure that you're linting and formatting your code to avoid problematic code and conform to a repo-wide style.

This is due to Prettier; it automatically wraps lines based on their length. Assuming nothing else is going on, the code formatter running on commit is intended functionality.

File: modules/planner/hooks/newPlanFlow.ts [hunk 2]
```diff
@@ -73,12 +73,13 @@ export function useCreateNewPlanFlow(
       case 'SELECT_ADDITIONS':
         setPlanState('TRANSFER_COURSES');
         break;
-      case 'FINALIZE_PLAN_GENERATION':
-        setPlanState('SELECT_ADDITIONS');
-        break;
-      case 'DONE':
+        // case 'FINALIZE_PLAN_GENERATION':
+        //   setPlanState('SELECT_ADDITIONS');
+        //   break;
```
Comment (CharlieMahana) [inline modules/planner/hooks/newPlanFlow.ts:78]: Either remove or uncomment this code

**Comment by WillieCubed** [comment]
#### 📝 Changed routes:


| [**…rding_Pages/disclaimer**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/disclaimer) | [**…ding_Pages/disclaimers**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/disclaimers) |
|:-:|:-:|
|<a href="https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/disclaimer">Screenshot of /Onboarding_Pages/disclaimer</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/CWscEpNptJLxPldDdKHYegNpPcn95cFvGRA7Qvnt71Q0s4VCO0vnI-0nvdCfMc0OEJ7F-BkY4-45ezd7fmdJTQ.png">(view full size)</a> |<a href="https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/disclaimers">Screenshot of /Onboarding_Pages/disclaimers</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/fFTx8irDf6QWtzVvb5qCFbAztgtM23_rn8OwB_w84WMn_BrY07xtP9r8dH5DxXWO1VG8m_kJjIVL7JLpUyeOWg.png">(view full size)</a> |



| [**/Onboarding_Pages/pg_1**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/pg_1) | [**/Onboarding_Pages/pg_2**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/pg_2) |
|:-:|:-:|
|<a href="https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/pg_1">Screenshot of /Onboarding_Pages/pg_1</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/1LnPlP2ClwD8qH8M-O_1QICRGZNBvac-A_EW_onWcqevwjMkMoMQI0RB5ahZwuY4fQMTBQ4SjxqBMpSUQ5gc8g.png">(view full size)</a> |<a href="https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/pg_2">Screenshot of /Onboarding_Pages/pg_2</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/GWOCPmINyK9LPRQJtrRmGwhr1CCSz2OP6_7EEAKJQAYj_T49VUikc8GkVjVMjM0RQmG3spkpf4VLMqZVvtVJNA.png">(view full size)</a> |



| [**/Onboarding_Pages/pg_3**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/pg_3) | [**/Onboarding_Pages/privacy**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/privacy) |
|:-:|:-:|
|<a href="https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/pg_3">Screenshot of /Onboarding_Pages/pg_3</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/xBf4deOH4RAwzwxtseL2lWfvNVzYnBBsn2CiNOF7GVNd-5POJCnC4y26XcVPCZgVw_s372ElbPrrHLT01GD0gg.png">(view full size)</a> |<a href="https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/privacy">Screenshot of /Onboarding_Pages/privacy</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/uT1xgK326YNHAAINjg4nptoGI33iaaW30Ry_Lg6zYx9D315VIvaNe5ExyilesAjrVC5oRri-BP1RJ1QgQpekLA.png">(view full size)</a> |



And 5 other routes:
- [**/Onboarding_Pages/welcome**](https://nebula-web-kc4pnnena-acmutd.vercel.app/Onboarding_Pages/welcome)
- [**/app/onboarding**](https://nebula-web-kc4pnnena-acmutd.vercel.app/app/onboarding)
- [**/app/plans/\[...planId\]**](https://nebula-web-kc4pnnena-acmutd.vercel.app/app/plans/[...planId])
- [**/app/plans**](https://nebula-web-kc4pnnena-acmutd.vercel.app/app/plans)
- [**/app/profile**](https://nebula-web-kc4pnnena-acmutd.vercel.app/app/profile)



Commit 961a6e62d75433071805c0135dae782d8263de2f (https://nebula-web-kc4pnnena-acmutd.vercel.app).

**Pull request merged**